src/microengine_utils/filesystem.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+import tempfile
+from contextlib import suppress
+
+import uuid
+import os
+import os.path
+from pathlib import Path, PureWindowsPath
+from typing import Union, Optional
+from .constants import VENDOR_DIR, PLATFORM_OS
+
+
+def as_wine_path(filename: 'str', *, check_exists=False) -> 'PureWindowsPath':  # noqa
+    """Converts a Unix path to the corresponding WinNT path"""
+    root, *rest = Path(filename).absolute().resolve().parts
+    return PureWindowsPath('Z:\\').joinpath(*(p.replace('/', '\\') for p in rest))
+
+
+async def winepath(path: 'os.PathLike', output='Windows') -> 'PureWindowsPath':
```

Comment:
We should have a sync version of this one, though.

Reply:
I was having questions about if we should even include this at all, it's only useful to WINE engines and so far none are using synchronous scanning. That said, there's something to be said for `microengine-utils` having a single definitive implementation of `winepath`.

I think we should include it as-is for future one offs, or just take it out entirely. It's up to you though.